pkg/istio/rbac/common/util_test.go

```diff
@@ -1018,3 +1018,185 @@ func getAuthzPolicyItem(action model.Event) Item {
 	item.Resource = out
 	return item
 }
+
+func TestParseComponentsEnabledAuthzPolicy(t *testing.T) {
+	type inputData struct {
+		description string
+		objectList  string
+	}
+	type outputData struct {
+		result *ComponentEnabled
+		err    string
+	}
+	type testData struct {
+		input  inputData
+		output outputData
+	}
+	tests := []testData{
+		{
+			input: inputData{
+				description: "Parse services-enabled-authzpolicy list",
+				objectList:  "namespace1/service1,namespace2/service2",
+			},
+			output: outputData{
+				result: &ComponentEnabled{
+					serviceList: []ServiceEnabled{
+						{
+							service:   "service1",
+							namespace: "namespace1",
+						},
+						{
+							service:   "service2",
+							namespace: "namespace2",
+						},
+					},
+					namespaceList: []string{},
+					cluster:       false,
+				},
+				err: "",
+			},
+		}, {
+			input: inputData{
+				description: "Services-enabled-authzpolicy list item has invalid format",
+				objectList:  "service1-namespace1,service2-namespace2",
+			},
+			output: outputData{
+				result: nil,
+				err:    "service item service1-namespace1 from command line arg components-enabled-authzpolicy is in incorrect format",
+			},
+		}, {
+			input: inputData{
+				description: "Parse namespaces-enabled-authzpolicy list",
+				objectList:  "ns1/*,ns2/*,ns3/*",
+			},
+			output: outputData{
+				result: &ComponentEnabled{
+					serviceList:   []ServiceEnabled{},
+					namespaceList: []string{"ns1", "ns2", "ns3"},
+					cluster:       false,
+				},
+				err: "",
+			},
+		}, {
+			input: inputData{
+				description: "Parse clusters-enabled-authzpolicy argument",
+				objectList:  "*",
+			},
+			output: outputData{
+				result: &ComponentEnabled{
+					serviceList:   []ServiceEnabled{},
+					namespaceList: []string{},
+					cluster:       true,
+				},
+				err: "",
+			},
+		},
+	}
+	for _, testcase := range tests {
+		components, err := ParseComponentsEnabledAuthzPolicy(testcase.input.objectList)
+		if err != nil {
+			if err.Error() != testcase.output.err {
+				t.Errorf("Wrong error message. Expected: %s, Actual: %s", testcase.output.err, err.Error())
+			} else {
+				continue
+			}
+		}
+
+		if len(components.serviceList) != len(testcase.output.result.serviceList) {
```

Comment:
I believe we can simplify all these checks using assert.EqualValues to do the comparisons here.